homeassistant/components/sensor/fibaro.py

```diff
@@ -0,0 +1,99 @@
+"""
+Support for Fibaro sensors.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.fibaro/
+"""
+import logging
+
+from homeassistant.const import (
+    DEVICE_CLASS_TEMPERATURE, DEVICE_CLASS_HUMIDITY,
+    DEVICE_CLASS_ILLUMINANCE, TEMP_CELSIUS, TEMP_FAHRENHEIT)
+from homeassistant.helpers.entity import Entity
+from homeassistant.components.sensor import ENTITY_ID_FORMAT
+from homeassistant.components.fibaro import (
+    FIBARO_CONTROLLER, FIBARO_DEVICES, FibaroDevice)
+
+SENSOR_TYPES = {
+    'com.fibaro.temperatureSensor':
+        ['Temperature', None, None, DEVICE_CLASS_TEMPERATURE],
+    'com.fibaro.smokeSensor':
+        ['Smoke', 'ppm', 'mdi:fire', None],
+    'CO2':
+        ['CO2', 'ppm', 'mdi:cloud', None],
+    'com.fibaro.humiditySensor':
+        ['Humidity', '%', None, DEVICE_CLASS_HUMIDITY],
+    'com.fibaro.lightSensor':
+        ['Light', 'lx', None, DEVICE_CLASS_ILLUMINANCE]
+}
+
+DEPENDENCIES = ['fibaro']
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Fibaro controller devices."""
+    if discovery_info is None:
+        return
+
+    add_entities(
+        [FibaroSensor(device, hass.data[FIBARO_CONTROLLER])
+         for device in hass.data[FIBARO_DEVICES]['sensor']], True)
+
+
+class FibaroSensor(FibaroDevice, Entity):
+    """Representation of a Fibaro Sensor."""
+
+    def __init__(self, fibaro_device, controller):
+        """Initialize the sensor."""
+        self.current_value = None
+        self.last_changed_time = None
+        super().__init__(fibaro_device, controller)
+        self.entity_id = ENTITY_ID_FORMAT.format(self.ha_id)
+        if fibaro_device.type in SENSOR_TYPES:
+            self._device_class = SENSOR_TYPES[fibaro_device.type][3]
+            self._icon = SENSOR_TYPES[fibaro_device.type][2]
+            self._unit_of_measurement = SENSOR_TYPES[fibaro_device.type][1]
+        else:
+            self._icon = None
+            self._device_class = None
+            self._unit_of_measurement = None
```

Comment:
Please order the same way with if/else, and preferably in the same order as in `SENSOR_TYPES`. Makes it easier to read :)

Reply:
Done. Thanks!